### PR TITLE
feat: Allow contextType to specify an TypingImport config

### LINKED ISSUE #265: Find more seamless way (ideally zero-config) to register Context type
The following is the current way to get a typed context object using Nexus. This feels pretty cumbersome and I'd prefer a zero-config way (while still being transparent) to "register" the Context type.

image

### PR DESCRIPTION
**Pull request by tgriesser**
Closes #265

Allows specifying the `contextType` explicitly, similar to the `rootTyping` for an individual type.

### CODE REVIEW

File: src/builder.ts [hunk 1]
```diff
@@ -262,9 +263,13 @@ export interface TypegenInfo {
...
   backingTypeMap: { [K in GetGen<'objectNames'>]?: string }
   /**
-   * The type of the context for the resolvers
+   * The string type of the context for the resolvers
    */
   contextType?: string
+  /**
+   * Info about where to import the context from
+   */
+  contextTypeImport?: TypingImport
```
Comment (Weakky) [inline src/builder.ts:272]: Any reason to have two different fields? I feel like this is even more confusing as it's not clear which one you're supposed to pick.

Would you consider making `contextType` independent of `typegenAutoConfig.sources`, therefore only allowing `contextType: TypingImport`? From my experience, the context type is 99% of the case in a separate file/folder than where your models lives anyway.

Comment (tgriesser) [reply src/builder.ts:272]: > Any reason to have two different fields? I feel like this is even more confusing as it's not clear which one you're supposed to pick.

I thought this was only an internal type (`TypegenInfo` vs `TypegenAutoConfigOptions` which doesn't have this), so it didn't matter.

> Would you maybe consider making contextType entirely independent of typegenAutoConfig.sources, therefore only allowing contextType: TypingImport? From my experience, the context type is 99% of the case in a separate file/folder than where your models live anyway. It always felt weird to configure the context type as part of your backing/root types, which are arguably two different things.

Yeah that sounds reasonable as well!

Comment (jasonkuhrt) [reply src/builder.ts:272]: > Would you maybe consider making contextType entirely independent of typegenAutoConfig.sources, therefore only allowing contextType: TypingImport? From my experience, the context type is 99% of the case in a separate file/folder than where your models live anyway. It always felt weird to configure the context type as part of your backing/root types, which are arguably two different things.

+1

Comment (Weakky) [reply src/builder.ts:272]: > I thought this was only an internal type (TypegenInfo vs TypegenAutoConfigOptions which doesn't have this), so it didn't matter.

Oh, sorry, you're right I didn't look properly!

Comment (jasonkuhrt) [reply src/builder.ts:290]: Let's take this up in follow up PRs

**Pull request merged**